src/components/NavigationDrawer.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import { StackNavigationProp } from '@react-navigation/stack'
+import { StyleSheet, View, Image, TouchableOpacity } from 'react-native'
+import BottomSheet from 'reanimated-bottom-sheet'
+import Icon from 'react-native-vector-icons/FontAwesome5'
+
+import logo from '../../assets/logo.png'
+
+type NavigationBarProps = {
+    navigation: StackNavigationProp<>;
+  }
+
+const styles = StyleSheet.create( {
+  navigationPanel: {
+    height: '100%',
+    backgroundColor: '#f9f9fa',
+    opacity: 0.92,
+  },
+  navigationPanelHeader: {
+    backgroundColor: '#f9f9fa',
+    paddingTop: 7,
+    borderTopLeftRadius: 20,
+    borderTopRightRadius: 20,
+    alignItems: 'center',
+    opacity: 0.92,
+  },
+  navigationPanelHandle: {
+    width: 80,
+    height: 2,
+    borderRadius: 5,
+    backgroundColor: '#00000040',
+  },
+  navigationPanelIconContainer: {
+    flexDirection: 'row',
+    paddingTop: '5%',
+    paddingBottom: '10%',
+  },
+  navigationIconSpacing: {
+    paddingRight: 30,
+    paddingLeft: 30,
+  },
+  navigationPanelButtonContainer: {
+    flexDirection: 'row',
+    justifyContent: 'space-around',
+  },
+} )
+
+const Buttons = [
+  [ 'clock', 'WIP' ],
+  [ 'bookmark', 'WIP' ],
+  [ 'search', 'Search' ],
+  [ 'square', 'WIP' ],
+  [ 'ellipsis-h', 'WIP' ],
+]
+
+const NavigationBar: React.FC<NavigationBarProps> = ( { navigation } ) => {
+  const renderNavigationHeader = () => (
+    <View style={styles.navigationPanelHeader}>
+      <View style={styles.navigationPanelHandle} />
+      <View style={styles.navigationPanelIconContainer}>
+        {Buttons.map( ( [ iconName, destination ] ) => (
+          <View key={iconName} style={styles.navigationIconSpacing}>
+            <Icon
+              name={iconName}
+              size={20}
+              color="rgb(47,124,246)"
+              onPress={() => navigation.navigate( destination )}
+            />
+          </View>
+        ) )}
+      </View>
+    </View>
+  )
+
+  const renderNavigationContent = () => (
+    <View style={styles.navigationPanel}>
+      <View style={styles.navigationPanelButtonContainer}>
+        <Icon.Button name="arrow-left" size={30} borderRadius={15} />
+        <TouchableOpacity activeOpacity={0.5}>
+          <Image style={{ width: 60, height: 60 }} source={logo} />
```

Comment:
can we not make styling inline please

Reply:
This part is just placeholder so will be removed once we have content to put in there.